client/homebrew/editor/editor.jsx

```diff
@@ -82,11 +83,49 @@ const Editor = createClass({
 		if(!this.refs.codeEditor) return;
 		const codeMirror = this.refs.codeEditor.codeMirror;
 
+		//Clear all Text styles
+		const ranges=codeMirror.getAllMarks();
+		for (let i=0;i<ranges.length;i++) ranges[i].clear();
```

Comment:
Why this piece of code is needed? `range` is not used anywhere else

Reply:
This resets the styling in the codemirror Editor. If you type some code that gets highlighted (\page, \column, {{etc...) and then move those blocks around (press the enter key in a weird spot, delete stuff) such that the code is no longer valid, the style stays applied, so you end up with highlighted bits of code all over that don't clear up until you refresh. This just grabs the array of custom code styling and clears them out before re-applying the styles in the proper places. `ranges` was just taken from the example on the codeMirror blog, but we can rename it for clarity. Maybe `customHilighting` .

Reply:
Renamed this variable and rewrote comment for clarity.